boruta/boruta_py.py

```diff
@@ -279,7 +280,7 @@ def _fit(self, X, y):
                 self.estimator.set_params(n_estimators=n_tree)
 
             # make sure we start with a new tree in each iteration
-            self.estimator.set_params(random_state=self.random_state)
+            self.estimator.set_params(random_state=self.random_state.randint(0, 10000))
```

Comment:
will this not break all other estimators?

Reply:
I should test it extensively on all other estimators. Do you have a testing frame?

Reply:
alternatively, we could do a type check here and use your update if the estimator is a light gbm and leave the random state as is if it's not..

Reply:
that could be a solution, because at the moment LightGBM cannot work at all with Boruta since it is not fully scikit-learn complaint (but that's a pity because LightGBM is at the moment the most used GBM solution around)

Reply:
ok would you mind altering your PR to do just that and I'll merge then

Reply:
You could use an integer value, it'll run ok (see test in my PR) or detect when the estimator is a lightgbm and then not use a RandomState instance. 

```python
def _is_lightgbm(self):
    self.is_lgb = 'lightgbm' in str(type(self.estimator))
```

Related issue: Catboost doesn't allow to change the random seed. Here is a way to avoid the error but does not guarantee starting with a new tree (all details in the PR I submitted)

```python
def _is_catboost(self):
self.is_cat = 'catboost' in str(type(self.estimator))
# make sure we start with a new tree in each iteration
# Catboost doesn't allow to change random seed after fitting
if self.is_cat is False:
    self.estimator.set_params(random_state=self.random_state)
```

Reply:
> alternatively, we could do a type check here and use your update if the estimator is a light gbm and leave the random state as is if it's not..

I made the necessary changes to the pr :-)

Reply:
> You could use an integer value, it'll run ok (see test in my PR) or detect when the estimator is a lightgbm and then not use a RandomState instance.
> 
> ```python
> def _is_lightgbm(self):
>     self.is_lgb = 'lightgbm' in str(type(self.estimator))
> ```
> 
> Related issue: Catboost doesn't allow to change the random seed. Here is a way to avoid the error but does not guarantee starting with a new tree (all details in the PR I submitted)
> 
> ```python
> def _is_catboost(self):
> self.is_cat = 'catboost' in str(type(self.estimator))
> # make sure we start with a new tree in each iteration
> # Catboost doesn't allow changing random seed after fitting
> if self.is_cat is False:
>     self.estimator.set_params(random_state=self.random_state)
> ```

Thank you Thomas, I used the _is_lightgbm function that you suggested